components/mongodb/mongodb.app.mjs

```diff
@@ -1,11 +1,289 @@
+            const Model = mongoose.model(collection, this.getSchemaByData(data));
+            await Model.findByIdAndUpdate(_id, this.parseStrings(
+              data,
+              parseNumbers,
+              parseBooleans,
+              parseDates,
+            ));
+            mongoose.connection.close(() => {
+              setTimeout(() => resolve());
+            });
+          } catch (err) {
+            mongoose.connection.close(() => {
+              setTimeout(() => reject(err));
+            });
+          }
+        })
+          .catch((err) => reject(err));
+      });
+    },
+    deleteDocumentById(collection, _id) {
+      return new Promise((resolve, reject) => {
+        this.connect()
+          .then(async () => {
+            try {
+              const schema = new mongoose.Schema(undefined, {
+                strict: false,
+              });
+              const Model = mongoose.model(collection, schema);
+              await Model.deleteOne({
+                _id,
+              });
+              mongoose.connection.close(() => {
+                setTimeout(() => resolve());
+              });
+            } catch (err) {
+              mongoose.connection.close(() => {
+                setTimeout(() => reject(err));
+              });
+            }
+          })
+          .catch((err) => reject(err));
+      });
+    },
+    findDocumentById(collection, _id) {
+      return new Promise((resolve, reject) => {
+        this.connect()
+          .then(async () => {
+            try {
+              const schema = new mongoose.Schema(undefined, {
+                strict: false,
+              });
+              const Model = mongoose.model(collection, schema);
+              const document = await Model.findById(_id);
+              mongoose.connection.close(() => {
+                setTimeout(() => resolve(document));
+              });
+            } catch (err) {
+              mongoose.connection.close(() => {
+                setTimeout(() => reject(err));
+              });
+            }
+          })
+          .catch((err) => reject(err));
+      });
+    },
+    searchDocuments(collection, filter, skip, limit, parseNumbers, parseBooleans, parseDates) {
+      return new Promise((resolve, reject) => {
+        this.connect()
+          .then(async () => {
+            try {
+              const schema = new mongoose.Schema(undefined, {
+                strict: false,
+              });
+              const Model = mongoose.models[collection] || mongoose.model(collection, schema);
+              const documents = await Model.find(filter)
+                .skip(skip)
+                .limit(limit);
+              mongoose.connection.close(() => {
+                setTimeout(() => resolve(documents.map((doc) => this.parseStrings(
+                  doc._doc,
+                  parseNumbers,
+                  parseBooleans,
+                  parseDates,
+                ))));
+              });
```

Comment:
Sorry, my last comment was confusing. The issue I was having was that I wasn't able to search for records with numbers/booleans/dates because they would get converted to a string prior to running the search. 

For instance, I had a record with a value of 3, but the search filter would search for "3", and therefore not find anything. 

I think we need to parse the values of `filter` before they're put in `Model.find` so that it's possible to search for numbers/booleans/dates.
```suggestion
              const Model = mongoose.models[collection] || mongoose.model(collection, schema);
              const documents = await Model.find(
                this.parseStrings(filter, parseNumbers, parseBooleans, parseDates),
              )
                .skip(skip)
                .limit(limit);
              mongoose.connection.close(() => {
                setTimeout(() => resolve(documents));
              });
```